src/ExtUvLoop.php

```diff
@@ -0,0 +1,320 @@
+<?php
+
+namespace React\EventLoop;
+
+use React\EventLoop\Tick\FutureTickQueue;
+use React\EventLoop\Timer\Timer;
+use SplObjectStorage;
+
+/**
+ * An `ext-uv` based event loop.
+ *
+ * This loop uses the [`uv` PECL extension](https://pecl.php.net/package/uv),
+ * that provides an interface to `libuv` library.
+ *
+ * This loop is known to work with PHP 7+.
+ *
+ * @see https://github.com/bwoebi/php-uv
+ */
+final class ExtUvLoop implements LoopInterface
+{
+    private $uv;
+    private $futureTickQueue;
+    private $timers;
+    private $streamEvents = array();
+    private $readStreams = array();
+    private $writeStreams = array();
+    private $running;
+    private $signals;
+    private $signalEvents = array();
+    private $streamListener;
+
+    public function __construct()
+    {
+        if (!function_exists('uv_loop_new')) {
```

Comment:
Should be updated to use global references as per https://github.com/reactphp/event-loop/pull/167 that was merged in the meantime.

Reply:
Fixed :+1: